src/main/java/ChatBot.java

```diff
@@ -0,0 +1,158 @@
+            String command = in.nextLine().trim();
+
+            try {
+                if (command.equalsIgnoreCase("bye")) {
+                    printFormattedMessage("Bye. Hope to see you again soon!");
+                    break;
+                } else if (command.equalsIgnoreCase("list")) {
+                    printTaskList();
+                } else if (command.startsWith("mark ")) {
+                    markTaskAsDone(command.substring(5));
+                } else if (command.startsWith("unmark ")) {
+                    unmarkTaskAsDone(command.substring(7));
+                } else if (command.startsWith("todo ")) {
+                    addTask(command.substring(5), "todo");
+                } else if (command.startsWith("event ")) {
+                    addTask(command.substring(6), "event");
+                } else if (command.startsWith("deadline ")) {
+                    addTask(command.substring(9), "deadline");
+                } else {
+                    printFormattedMessage("Unknown command. Please enter 'todo', 'event', " +
+                            "'deadline', 'list', 'mark', 'unmark', or 'bye'.");
+                }
+            } catch (ChatBotExceptions e) {
+                printFormattedMessage(e.getMessage());
+            }
+        }
+    }
+
+
+    private void addTask(String taskDescription, String taskType) throws ChatBotExceptions {
+        Task newTask;
+
+        if (taskType.equals("todo")) {
+            newTask = new ToDos(taskDescription);
+        } else if (taskType.equals("deadline")) {
+            String[] parts = taskDescription.split(" by ");
+            if (parts.length < 2) {
+                throw new ChatBotExceptions("Invalid deadline format. " +
+                        "Please use 'deadline <description> by <date>'.");
+            }
+            String description = parts[0];
+            String byDate = parts[1];
+            newTask = new Deadlines(description, byDate);
+        } else if (taskType.equals("event")) {
+            String[] parts = taskDescription.split(" from | to ");
+            if (parts.length < 3) {
+                throw new ChatBotExceptions("Invalid event format. " +
+                        "Please use 'event <description> from <start> to <end>'.");
+            }
+            String description = parts[0];
+            String fromDate = parts[1];
+            String toDate = parts[2];
+            newTask = new Events(description, fromDate, toDate);
+        } else {
+            throw new ChatBotExceptions("Invalid command.");
+        }
+
+        tasks.add(newTask);
+        System.out.println("Got it. I've added this task:");
+        System.out.println("   " + newTask.getStatusIcon() + " " + newTask.getDescription());
+        printFormattedMessage(" Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    private void printTaskList() {
+
+        String horizontalLines = "____________________________________________________________";
```

Comment:
it would be better to make this a class-level attribute and declare it as a constant ?